pkg/controllers/jobset_controller.go

```diff
@@ -675,8 +675,12 @@ func labelAndAnnotateObject(obj metav1.Object, js *jobset.JobSet, rjob *jobset.R
 
 	// Only set exclusive key label/annotation on jobs/pods if the parent JobSet
 	// is using exclusive placement.
-	if _, exists := js.Annotations[jobset.ExclusiveKey]; exists {
-		annotations[jobset.ExclusiveKey] = js.Annotations[jobset.ExclusiveKey]
+	if topologyDomain, exists := rjob.Template.Annotations[jobset.ExclusiveKey]; exists {
+		annotations[jobset.ExclusiveKey] = topologyDomain
+		// Add nodeSelectorStrategy annotation if it is being used.
+		if value, ok := rjob.Template.Annotations[jobset.NodeSelectorStrategyKey]; ok {
+			annotations[jobset.NodeSelectorStrategyKey] = value
+		}
```

Comment:
is this a bug?

Reply:
if so, this is a reason to patch back.

Reply:
No, previously we just were only setting the nodeSelectorStrategy annotation at the JobSet level. Now that we are supporting exclusive placement strategy at the ReplicatedJob level, the nodeSelectorStrategy annotation also needs to be able to be set on the ReplicatedJob Job template, since the user may be using the nodeSelectorStrategy implementation of exclusive placement for this particular ReplicatedJob, not the entire JobSet.